packages/pwa-kit-react-sdk/src/ssr/universal/contexts/index.js

```diff
@@ -5,9 +5,41 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
-import React from 'react'
+import React, {useEffect} from 'react'
+import PropTypes from 'prop-types'
+import {withRouter} from 'react-router-dom'
+import {v4 as uuid} from 'uuid'
+
 import DeviceContext from '../device-context'
 
 const ExpressContext = React.createContext()
 
-export {DeviceContext, ExpressContext}
+const CorrelationIdContext = React.createContext()
+
+const CorrelationIdProvider = ({children, correlationId, location}) => {
+    const [id, setId] = React.useState(correlationId)
```

Comment:
Could you do something like this instead? 

`const [id, setId] = React.useSate(correlationId || uuid())`

This way we'll always have a uuid if one isn't provided, and it means we don't have to import the `uuid` module in multiple places, maybe it easier to swap out for something else if were to use a different lighter weight uuid generator.